build-system/app.js

```diff
@@ -751,13 +751,7 @@ app.get(['/examples/*.html', '/test/manual/*.html'], (req, res, next) => {
   });
 });
 
-function escapeRegExp(string) {
-  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
-}
-
 function elementExtractor(tagName, type) {
-  tagName = escapeRegExp(tagName);
-  type = escapeRegExp(type);
```

Comment:
To avoid the vulnerability, you'll have to escape the strings before creating the regex. Maybe you can do so without breaking `elementExtractor` by modifying line 755 in `escapeRegExp`? The original is from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions (scroll half way down the page)

Reply:
Or maybe you can replace the function I wrote with `escapeRegExp()` from the `lodash` package?

See https://lgtm.com/rules/2025670640/

Reply:
Src code here: https://github.com/lodash/lodash/blob/master/escapeRegExp.js

Reply:
In this particular case, to fix the breakage while remaining secure, it is sufficient just to escape the `type` parameter, as that is the only user-controlled value. The `tagName` parameter is one of the safe and hardcoded partial regexes: `"(amp-ad|amp-embed)"` or `"amp-analytics"`.

```javascript
function elementExtractor(tagName, type) {	
  type = escapeRegExp(type);
  return new RegExp(
    `<${tagName} [^>]*['"]${type}['"][^>]*>([\\s\\S]+?)</${tagName}>`,
       'gm');
}
```
--

Engineer at lgtm.com